website/profile/views.py

```diff
@@ -85,20 +86,64 @@ def update_user(auth):
     data = request.get_json()
 
     # TODO: Expand this to support other user attributes
-    if 'timezone' in data:
-        if data['timezone']:
-            user.timezone = data['timezone']
+
+    if 'emails' in data:
+        # removals
+        removed_emails = [
+            each
+            for each in user.emails + user.unconfirmed_emails
+            if each not in [x['address'] for x in data['emails']]
+        ]
+
+        for address in removed_emails:
+            if address in user.emails:
+                user.emails.remove(address)
+            user.remove_unconfirmed_emails(address)
+
+        # additions
+        added_emails = [
+            each['address']
+            for each in data['emails']
+            if each['address'] not in user.emails
+            and each['address'] not in user.unconfirmed_emails
+        ]
+
+        for address in added_emails:
+            try:
+                user.add_unconfirmed_email(address)
+            except ValueError:
```

Comment:
I was looking for that!

I changed it; it was raising `ValueError` on validation failure. When I refactored to a separate validator, I changed it to raise `ValidationError` instead. I knew I was catching it somewhere, but couldn't find where. Thanks! :bow: